### PR TITLE
REF: remove libreduction.SeriesGrouper

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

cc @bashtage per comment https://github.com/pandas-dev/pandas/pull/37468#discussion_r706100798 I'm taking your word for it that the change this makes in the crosstab test is correct.  Does this represent a bugfix?

### CODE REVIEW

File: pandas/tests/reshape/test_crosstab.py [hunk 1]
```diff
@@ -811,7 +811,7 @@ def test_categoricals(a_dtype, b_dtype):
     a_is_cat = is_categorical_dtype(a.dtype)
     assert not a_is_cat or a.value_counts().loc[1] == 0
     result = crosstab(a, b, margins=True, dropna=False)
-    values = [[18, 16, 34], [0, 0, np.nan], [34, 32, 66], [52, 48, 100]]
+    values = [[18, 16, 34], [0, 0, 0], [34, 32, 66], [52, 48, 100]]
```
Comment (jreback) [inline pandas/tests/reshape/test_crosstab.py:814]: may want to add a note for this (i agree that this looks more correct)

**Review by jreback — APPROVED** [review]
awesome!

**Comment by bashtage** [comment]
I agree that this looks like a bug fix too.

**Comment by jbrockmendel** [comment]
@bashtage can i pawn off the writing of a whatsnew on you?  i have zero experience with crosstab

**Comment by bashtage** [comment]
What's new in #43548

**Pull request merged**